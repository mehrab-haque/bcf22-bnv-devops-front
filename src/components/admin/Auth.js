import React, {useRef, useState} from 'react'
import {
    Button,
    Dialog,
    DialogActions,
    DialogContent,
    DialogTitle,
    Grid,
    LinearProgress,
    TextField
} from "@mui/material";
import {showToast} from "../../App";

const Auth=props=>{

    const loginRef=useRef();
    const passRef=useRef();

    const [loading,setLoading]=useState(false)

    const loginClick=()=>{
        setLoading(true)
        setTimeout(()=>{
            setLoading(false)
            props.login()
        }, 2000);
    }


    return(
        <div>
            <Dialog open={true}>
                {
                    loading?(
                        <LinearProgress/>
                    ):(
                        <div/>
                    )
                }
                <DialogTitle>
                    Admin Login
                </DialogTitle>
                <DialogContent>
                    <Grid container spacing={1}>
                        <Grid item xs={12}>
                            <TextField
                                inputRef={loginRef}
                                fullWidth
                                variant={'outlined'}
                                label={'Login'}
                                />
                        </Grid>
                        <Grid item xs={12}>
                            <TextField
                                inputRef={passRef}
                                fullWidth
                                type={'password'}
                                variant={'outlined'}
                                label={'Password'}
                            />
                        </Grid>
                    </Grid>
                </DialogContent>
                <DialogActions>
                    <Button
                        color={'primary'}
                        onClick={loginClick}
                        disabled={loading}
                        >
                        Login
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    )
}

export default Auth