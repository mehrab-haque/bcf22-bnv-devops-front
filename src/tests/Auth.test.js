import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Auth from "../components/admin/Auth";

describe("Component Auth Render", () => {
    it("renders without crashing given the required props", () => {
        const wrapper = shallow(<Auth/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
