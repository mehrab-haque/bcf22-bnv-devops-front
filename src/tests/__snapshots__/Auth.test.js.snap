// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Auth Render renders without crashing given the required props 1`] = `
<div>
  <ForwardRef(Dialog)
    open={true}
  >
    <div />
    <ForwardRef(DialogTitle)
      className="end-to-end-auth-selector"
    >
      Admin Login
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogContent)>
      <ForwardRef(Grid)
        container={true}
        spacing={1}
      >
        <ForwardRef(Grid)
          item={true}
          xs={12}
        >
          <ForwardRef(TextField)
            fullWidth={true}
            inputRef={
              Object {
                "current": undefined,
              }
            }
            label="Login"
            variant="outlined"
          />
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          xs={12}
        >
          <ForwardRef(TextField)
            fullWidth={true}
            inputRef={
              Object {
                "current": undefined,
              }
            }
            label="Password"
            type="password"
            variant="outlined"
          />
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button)
        color="primary"
        disabled={false}
        onClick={[Function]}
      >
        Login
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</div>
`;
